Makefile

```
@@ -1,42 +1,63 @@
-TESTS_IMAGE=packit-tests
+TEST_IMAGE=packit-tests
 
 CONTAINER_ENGINE ?= $(shell command -v podman 2> /dev/null || echo docker)
-TESTS_RECORDING_PATH=tests_recording
-TESTS_TARGET ?= ./tests/unit ./tests/integration ./tests/functional
+TEST_RECORDING_PATH=tests_recording
+TEST_TARGET ?= ./tests/unit ./tests/integration ./tests/functional
 CONTAINER_RUN_WITH_OPTS=$(CONTAINER_ENGINE) run --rm -ti -v $(CURDIR):/src --security-opt label=disable
-CONTAINER_TEST_COMMAND=bash -c "pip3 install .; make check"
+CONTAINER_TEST_COMMAND=bash -c "pip3 install .; make check-local"
+IS_FEDORA := $(shell uname -r | grep -c "\.fc[1-9]")
+ifeq ($(IS_FEDORA), 1)
+    PACKAGE_CHECKER := rpm -q --quiet
+    PACKAGE_INSTALLER := dnf install
+else
+    PACKAGE_CHECKER := :
+    PACKAGE_INSTALLER := :
+endif
 
 # In case you don't want to use pre-built image
 image:
 	$(CONTAINER_ENGINE) build --rm --tag $(IMAGE) .
 
-tests_image:
-	$(CONTAINER_ENGINE) build --rm --tag $(TESTS_IMAGE) -f Dockerfile.tests .
+tests-image:
+	$(CONTAINER_ENGINE) build --rm --tag $(TEST_IMAGE) -f Dockerfile.tests .
 	sleep 2
 
-tests_image_remove:
-	$(CONTAINER_ENGINE) rmi $(TESTS_IMAGE)
+tests-image-remove:
+	$(CONTAINER_ENGINE) rmi $(TEST_IMAGE)
 
 install:
 	pip3 install --user .
 
-check:
+prepare-check:	prepare-check-local
+
+check:	check-local
+
+prepare-check-local:
+	if ! $(PACKAGE_CHECKER) python3-pytest ; then sudo $(PACKAGE_INSTALLER) python3-pytest ; fi
+	if ! $(PACKAGE_CHECKER) python3-requre ; then sudo $(PACKAGE_INSTALLER) python3-requre ; fi
+	if ! $(PACKAGE_CHECKER) python3-flexmock ; then sudo $(PACKAGE_INSTALLER) python3-flexmock ; fi
+
+check-local:
 	find . -name "*.pyc" -exec rm {} \;
-	PYTHONPATH=$(CURDIR) PYTHONDONTWRITEBYTECODE=1 python3 -m pytest --verbose --showlocals $(TESTS_TARGET)
+	PYTHONPATH=$(CURDIR) PYTHONDONTWRITEBYTECODE=1 python3 -m pytest --verbose --showlocals $(TEST_TARGET)
+
+requre-data-cleanup:
+	requre-patch purge --replaces "copr.v3.helpers:login:str:somelogin" --replaces "copr.v3.helpers:token:str:sometoken" $(TEST_RECORDING_PATH)/test_data/*/*yaml
+	requre-patch create-symlinks $(CURDIR)/$(TEST_RECORDING_PATH)/test_data/
 
-requre_data_cleanup:
-	requre-patch purge --replaces "copr.v3.helpers:login:str:somelogin" --replaces "copr.v3.helpers:token:str:sometoken" $(TESTS_RECORDING_PATH)/test_data/*/*yaml
-	requre-patch create-symlinks $(CURDIR)/$(TESTS_RECORDING_PATH)/test_data/
+# example: TEST_TARGET=tests/unit/test_api.py make check-in-container
+prepare-check-in-container:
+	if ! $(PACKAGE_CHECKER) podman ; then sudo $(PACKAGE_INSTALLER) podman ; fi
 
-# example: TESTS_TARGET=tests/unit/test_api.py make check_in_container
-check_in_container: tests_image
-	$(CONTAINER_RUN_WITH_OPTS) --env TESTS_TARGET $(TESTS_IMAGE) $(CONTAINER_TEST_COMMAND)
+# example: TESTS_TARGET=tests/unit/test_api.py make check-in-container
+check-in-container: tests-image
+	$(CONTAINER_RUN_WITH_OPTS) --env TESTS_TARGET $(TEST_IMAGE) $(CONTAINER_TEST_COMMAND)
```

This also needs to be singular: 

```suggestion
	$(CONTAINER_RUN_WITH_OPTS) --env TEST_TARGET $(TEST_IMAGE) $(CONTAINER_TEST_COMMAND)
```

---

Thanks for review... changes incorporated

---

in `CONTRIBUTING.md` as well

---

Just took care of CONTRIBUTING.md - thanks for calling that to my attention!